### PR TITLE
54 move binop functions in ast

### LINKED ISSUE #54: Move binop functions in AST
# Description

During annotation time, we place the evaluation function for binary operators in the AST. This is not ideal since the AST should easily be serializable which is not possible with functions. Thus the we should postpone this function attaching until we construct the actual expression.

# Sub-tasks

- [x] Move the binop eval function to Expression construction

### PR DESCRIPTION
**Pull request by maanpalle**
# Pull Request

## Description

Removed the functions from the ast and cleaned up binaryoperator

Closes #54 

## Changes

1. Removed the functions from the ast
2. Moved binOpMap to seperate file

## Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

## How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

**Review by FBroy — APPROVED** [review]
This cleans up the code quite a bit, good job!

File: src/expressions/operators/arithmetic/BinaryEvaluationFunctionMap.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
+import { ValueType } from '../../../expressions/dataTypes/Value';
```
Comment (HectorPeeters) [inline src/expressions/operators/arithmetic/BinaryEvaluationFunctionMap.ts:1]: Good idea to move this into a separate file

File: src/parsing/compileAstToExpression.ts [hunk 2]
```diff
@@ -364,7 +366,22 @@ function binaryOperator(ast: IAST, compilationOptions: CompilationOptions) {
...
+		astHelper.followPath(ast, ['secondOperand', '*']),
+		'type'
+	) as SequenceType;
+	let firstType;
+	let secondType;
+	let evaluateFunction;
+	if (first && second) {
+		firstType = first.type;
+		secondType = second.type;
+		evaluateFunction = generateBinaryOperatorFunction(kind, firstType, secondType)[0];
```
Comment (HectorPeeters) [inline src/parsing/compileAstToExpression.ts:383]: This is indeed probably the easiest approach for now

**Review by HectorPeeters — APPROVED** [review]
Looks good to me!

**Pull request merged**